### ISSUE TITLE
No data for non-webextension add-ons in BigQuery

### ISSUE THREAD

**Issue by willdurand** [mozilla/addons-server#14533]
**Update:** it looks like we only have data for webextensions in BigQuery, nothing for langpacks, dictionaries and static themes and that's because Telemetry pings for those types are either not sent or different.

---

We won't have telemetry data for langpacks according to @jvillalobos so we'll need to take that into account in #14428 but also when we display a link to the stats pages in Dev Hub (and in addons-frontend but I'll file a different issue).

addons-frontend issues:

- hiding the average number of users in detail pages (langpacks only): https://github.com/mozilla/addons-frontend/issues/9459
- hiding the link to the stats in detail pages (langpacks only): https://github.com/mozilla/addons-frontend/issues/9460

**Comment by willdurand** [comment]
It turns out that we do have data for langpacks: https://sql.telemetry.mozilla.org/queries/71787/source. Numbers are surprisingly low, though. Closing this issue for now, though.

**Comment by willdurand** [comment]
Let's reopen this issue because obviously we've weird telemetry data for langpacks. This issue will actually apply to all non-webextension add-ons.

We have ideas to get data for langpacks and themes, but nothing for dictionaries yet.

**Comment by willdurand** [comment]
I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1644088

**Comment by willdurand** [comment]
@jvillalobos @diox @muffinresearch we need a long-term plan for non-webextensions. Given what we have in Telemetry right now, without a FF change, we won't have reliable data for langpacks and dictionaries IMO. Themes might be tricky too (cf. issue above) but at least, we do collect data already.

Questions:

1. What should we do for langpacks? Do we care about average daily users for these add-ons?

An option would be to not show the number of users anymore (in addons-frontend) and to remove stats dashboards for langpacks.

2. What should we do for dictionaries? I wasn't able to find any telemetry data for dictionaries. If we want to have stats for dicts, we'll need to make a change to FF (I believe). Are we OK with this? If so, we'll need to file a bug and get it prioritized.

We could also treat dictionaries like langpacks and apply the same suggestion as above. Assuming we would go with this suggestion, we would only have average daily users and stats dashboards for webextensions and themes.

Otherwise, a potential workaround would be to use download stats for the average daily users of langpacks and dictionaries. Given that we also want to get rid of the old download stats in Q3, this does not seem optimal because it only delays the actual problem. In any case, we'd still need to think about what to show in the usage stats pages.

Note for later: we'll need to think carefully about the download stats and make sure that we'll have all the data we need, for all add-ons and no matter which data source we'll use.

**Comment by jvillalobos** [comment]
Bug 1644088 covers themes, and to sum it up we do want to track those stats.

Long-term, we want language packs to be moved out of AMO and be treated as a Firefox feature, and we want dictionaries to be treated as regular extensions on AMO. So, with that in mind, I would like dictionaries to be counted for stats like we do extensions, while language packs I'm less concerned about.

Worst case scenario, I think we can change language packs and dictionaries to use Downloads as their public stat instead of Users, like we did with Search Engines before we discontinued them. Ideally we should show telemetry-based usage counts for both, but I don't think we should block on either being available.
